library/core/src/main/java/com/google/android/exoplayer2/text/cea/Cea608Decoder.java

```diff
@@ -284,6 +326,24 @@ protected void decode(SubtitleInputBuffer inputBuffer) {
       // If we've reached this point then there is data to process; flag that work has been done.
       captionDataProcessed = true;
 
+      if ((ccDataHeader & CC_VALID_FLAG) != CC_VALID_FLAG) {
+        // The screen needs to be cleared when the closed captions are turned off
+        // (the encoder flips the validity bit)
+        resetCueBuilders();
+        continue;
+      }
+
+      // When transmission is digital and reliable (TCP) we should not meet any data corruption.
+      // But any over-the-air content and test streams might give the decoder corrupted data, so we
+      // need to drop those. Every data byte should have odd parity.
+      // When corrupt data is met the screen needs to be cleared, so we:
+      // - reset the cue builders
```

Comment:
Please address comment on previous pull request about whether resetting is the best thing to do here.

Reply:
I agree, there are conflicting ideas in the legal documents about CEA608. This is from the Sarnoff compliance test document: 
> When the “NTSC bad parity test” is active, CC1, CC3 and CS1 should not display new captions. CC1and CC3 captions should persist on the screen for approximately one half second of this test condition, then the captions should clear.

In this case the test starts with correct data, then with bad parity after one half second, and they require the screen to be cleared.

Reply:
Checking the history of these changes suggest that we did not clear the screen in the first few iterations. And it was not leading previous captions staying on the screen too long. The original 608 documents suggested something like (this is my interpretation of the idea from the 80s)

1.  if the a command is failing the parity check: skip the command, wait for the repetition.
1. if a repeated command is failing the parity check: decoder acted upon the first one.
1. if the character fails the parity check, insert a placeholder character. 
1. if the stream is repeatedly failing the parity check, disable the decoding.

With the current over-the-air content and reliable transmission the only one still happening is the last case. The entire stream seems to have random data and we should not let the last caption stay on screen forever. Shall I add a TODO to add additional conditions later?